src/signature.sh

```diff
@@ -0,0 +1,348 @@
+
+        register_signature_operation 'ACKED' "$2"
+        if [[ "$?" -gt 0 ]]; then
+          options_values['ERROR']='You must configure your user.name and user.email with git '
+          options_values['ERROR']+='to use --add-acked-by or -a without an argument'
+          return 22 # EINVAL
+        fi
+
+        [[ ! "$2" ]] || shift
+        shift
+        ;;
+      --add-fixes | -f)
+        if [[ -n "${options_values['ADD_FIXES']}" ]]; then
+          options_values['ERROR']='Can not repeat option --add-fixes or -f'
+          return 22 # EINVAL
+        elif [[ ! "$2" ]]; then
+          options_values['ERROR']='The option --add-fixes or -f demands an argument'
+          return 22 # EINVAL
+        fi
+
+        options_values['ADD_FIXES']=1
+        options_values['FIXES_LINE']="$(format_fixes_message "$(str_strip "$2")")"
+
+        if [[ "$?" -gt 0 ]]; then
+          options_values['ERROR']="Invalid commit reference with --add-fixes or -f: ${2}"
+          return 22 # EINVAL
+        fi
+
+        options_values['FIXES_LINE']="${TRAILER_TAGS['FIXES']} ${options_values['FIXES_LINE']}"
+        trailer_operations+=('ADD_FIXES')
+        shift 2
+        ;;
+      --verbose)
+        options_values['VERBOSE']=1
+        shift
+        ;;
+      --)
+        # End of options, beginning of arguments.
+        # Overwrite default value if at least one argument is given.
+        [[ -n "$2" ]] && options_values['PATCH_OR_SHA']=''
+        shift
+        ;;
+      *)
+        # Get all passed arguments each loop
+        if [[ "$1" == *"*"* ]]; then
+          # Expand the glob and loop through each resulting file
+          for arg in $1; do
+            options_values['PATCH_OR_SHA']+=" $arg"
+          done
+        else
+          options_values['PATCH_OR_SHA']+=" $1"
+        fi
+        shift
+        ;;
+    esac
+  done
+}
+
+function signature_help()
+{
+  if [[ "$1" == --help ]]; then
+    include "${KW_LIB_DIR}/help.sh"
+    kworkflow_man 'handle-trailer'
+    return
+  fi
+  printf '%s\n' 'kw handle-trailer:' \
+    '  handle-trailer (--add-signed-off-by | -s) (<name>) [<patchset> | <sha>] - Add Signed-off-by' \
+    '  handle-trailer (--add-reviewed-by | -r) (<name>) [<patchset> | <sha>] - Add Reviewed-by' \
+    '  handle-trailer (--add-acked-by | -a) (<name>) [<patchset> | <sha>] - Add Acked-by' \
+    '  handle-trailer (--add-fixes | -f) [<fixed-sha>] [<patchset> | <sha>] - Add Fixes' \
+    '  handle-trailer (--verbose) - Show a detailed output'
+}
+
+# This function registers a signature operation into
+# trailer_operations that will be properly excuted later.
+# It attempts to use the user's name and email configured
+# if a signature is not passed as argument and gives
+# an error if they are not properly set with git config.
+#
+# @keyword Holds a string that is a keyword for a specific operation.
+# @signature Holds a string like 'NAME <EMAIL>' defining the signature.
+function register_signature_operation()
```

Comment:
I don't think this is a good name. Basically, you are formatting the input, so maybe you can use a name that better describes this.